apps/intellij/src/main/kotlin/dev/nx/console/generate/NxReMoveProjectDialog.kt

```diff
@@ -0,0 +1,201 @@
+                            } else {
+                                text(reMoveGenerators.get(0))
+                            }
+                        }
+                        .layout(RowLayout.PARENT_GRID)
+                    row {
+                            label("Destination:")
+
+                            appsOrLibsText = text("").gap(RightGap.SMALL)
+                            updateDestinationDirHint(model.project)
+
+                            destinationField =
+                                textField()
+                                    .bindText(model::directory)
+                                    .horizontalAlign(HorizontalAlign.FILL)
+                                    .component
+                        }
+                        .visible(mode == "move")
+                        .layout(RowLayout.PARENT_GRID)
+                }
+                .withPreferredWidth(400)
+        this.model = model
+        return panel
+    }
+
+    override fun doValidateAll(): MutableList<ValidationInfo> {
+        val validationList = mutableListOf<ValidationInfo>()
+        panel.apply()
+        if (model.project == "") {
+            validationList.add(ValidationInfo("Project is required", projectField))
+        }
+        if (mode == "move" && model.directory == "") {
+            validationList.add(ValidationInfo("Destination is required", destinationField))
+        }
+        return validationList
+    }
+
+    override fun createActions(): Array<Action> {
+        return arrayOf(cancelAction, DryRunAction() { -> dryRunCallback(this) }, okAction)
+    }
+
+    // for some reason kotlin can't resolve DialogWrapperAction
+    protected class DryRunAction(val callback: () -> Unit) : AbstractAction("Dry Run") {
+        override fun actionPerformed(e: ActionEvent?) {
+            callback()
+        }
+    }
+    private fun getShortcutHint(): String {
+        return "Use ${
+          KeymapUtil.getFirstKeyboardShortcutText(
+        ActionManager.getInstance().getAction(IdeActions.ACTION_CODE_COMPLETION)
+      )} for project completion\n"
+    }
+
+    fun updateDestinationDirHint(projectType: String?) {
+        val appOrLibDirByProjectType = getDestinationDirHint(projectOptions?.get(projectType))
+        appsOrLibsText.let {
+            if (it == null) {
+                return
+            }
+            if (appOrLibDirByProjectType == "") {
+                it.visible(false)
+                return
+            }
+            it.visible(true)
+            it.component.text = appOrLibDirByProjectType
+        }
+    }
+
+    private fun getDestinationDirHint(projectType: String?): String {
+        if (workspaceLayout == null) {
+            return ""
+        }
+        if (projectType == "application") {
+            return workspaceLayout.appsDir.let {
+                if (it == null || it == "") {
+                    return ""
+                }
+                "${it}/"
+            }
```

Comment:
We could probably use the `isNullOrEmpty` function to do this check as well.
```suggestion
            return workspaceLayout.appsDir.let {
                if (it.isNullOrEmpty()) {
                    return ""
                }
                "${it}/"
            }
```